File: src/main/java/me/nylestroke/nylemod/event/ModPlayerEventCopyFrom.java
package me.nylestroke.nylemod.event;

import me.nylestroke.nylemod.util.IEntityDataSaver;
import net.fabricmc.fabric.api.entity.event.v1.ServerPlayerEvents;
import net.minecraft.server.network.ServerPlayerEntity;

public class ModPlayerEventCopyFrom implements ServerPlayerEvents.CopyFrom {

    @Override
    public void copyFromPlayer(ServerPlayerEntity oldPlayer, ServerPlayerEntity newPlayer, boolean alive) {
        IEntityDataSaver original = ((IEntityDataSaver) oldPlayer);
        IEntityDataSaver player = ((IEntityDataSaver) newPlayer);

        player.getPersistentData().putIntArray("homePos", original.getPersistentData().getIntArray("homePos"));
    }

}

File: src/main/kotlin/me/nylestroke/nylemod/event/ModPlayerEventCopyFrom.kt
package me.nylestroke.nylemod.event

import me.nylestroke.nylemod.util.IEntityDataSaver
import net.fabricmc.fabric.api.entity.event.v1.ServerPlayerEvents
import net.minecraft.server.network.ServerPlayerEntity

class ModPlayerEventCopyFrom : ServerPlayerEvents.CopyFrom {
    override fun copyFromPlayer(oldPlayer: ServerPlayerEntity, newPlayer: ServerPlayerEntity, alive: Boolean) {
        val original: IEntityDataSaver = oldPlayer as IEntityDataSaver
        val player: IEntityDataSaver = newPlayer as IEntityDataSaver
        player.getPersistentData().putIntArray("homePos", original.getPersistentData().getIntArray("homePos"))
    }
}